app/src/main/java/org/simple/clinic/scanid/ScanSimpleIdEffectHandler.kt

```diff
@@ -11,7 +12,8 @@ import org.simple.clinic.util.scheduler.SchedulersProvider
 class ScanSimpleIdEffectHandler @AssistedInject constructor(
     private val schedulersProvider: SchedulersProvider,
     private val patientRepository: PatientRepository,
-    @Assisted private val uiActions: ScanSimpleIdUiActions
+    @Assisted private val uiActions: ScanSimpleIdUiActions,
+    private val moshi: Moshi
```

Comment:
Instead of passing the `Moshi` object directly, can we instead have a wrapper that parses the data? That way we don't have to build an actual `Moshi` instance in tests and instead just use mocks. WDYT?

Reply:
There seem to be no problem in creating Moshi instances in tests. Just checked their Github. I don't think we need this then

Reply:
Hmm, @vinaysshenoy do you think it's fine for us to use Moshi directly in this scenario?

Reply:
Instead of Moshi, we can pass the adapter, maybe? This is what I generally do in this scenario. Apart from that, I think this is fine.

Reply:
A potential improvement can be have something like an `NHIDDataParser` which abstracts away the implementation detail that the scanned string is a JSON string, but I don't think it is important enough to hold up merging this PR in. Might be a future maintenance effort.